"assertion failed: NoType" with private type aliases inside a trait leaking

## Compiler version

crashes on both 3.3.0 and 3.5.0

## Minimized code

```Scala
object CompilerCrash {
  trait Scope {
    private type Event = String

    case class Cmd(events: List[Event])
  }

  new Scope {
    val commands = List(
      Cmd(List("1", "2"))
    )
  }
}
```

Also the same code fails to compile here: https://scastie.scala-lang.org/migesok/VCm2LTz1ToWFmuG5mYBblw/1

It compiles fine if `private` modifier is removed.

## Output (click arrow to expand)
<!--
Note: to get a full crash log when using scala-cli, run compilation with the flag `--server=false`
-->
<details>

```
[info] compiling 1 Scala source to /Users/myuser/prog/scala3-compiler-crash/target/scala-3.3.3/classes ...
[info]
[info]   exception occurred while typechecking /Users/myuser/prog/scala3-compiler-crash/src/main/scala/CompilerCrash.scala
[info]
[info]   An unhandled exception was thrown in the compiler.
[info]   Please file a crash report here:
[info]   https://github.com/lampepfl/dotty/issues/new/choose
[info]
[info]      while compiling: /Users/myuser/prog/scala3-compiler-crash/src/main/scala/CompilerCrash.scala
[info]         during phase: typer
[info]                 mode: Mode(ImplicitsEnabled)
[info]      library version: version 2.13.12
[info]     compiler version: version 3.3.3
[info]             settings: -classpath /Users/myuser/prog/scala3-compiler-crash/target/scala-3.3.3/classes:/Users/myuser/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala3-library_3/3.3.3/scala3-library_3-3.3.3.jar:/Users/myuser/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.13.12/scala-library-2.13.12.jar -d /Users/myuser/prog/scala3-compiler-crash/target/scala-3.3.3/classes
[error] ## Exception when compiling 1 sources to /Users/myuser/prog/scala3-compiler-crash/target/scala-3.3.3/classes
[error] java.lang.AssertionError: assertion failed: NoType
[error] scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
[error] dotty.tools.dotc.core.Types$TypeBounds.<init>(Types.scala:5179)
[error] dotty.tools.dotc.core.Types$RealTypeBounds.<init>(Types.scala:5254)
[error] dotty.tools.dotc.core.Types$TypeBounds$.apply(Types.scala:5298)
[error] dotty.tools.dotc.core.Types$TypeBounds.derivedTypeBounds(Types.scala:5186)
[error] dotty.tools.dotc.core.Types$TypeMap.derivedTypeBounds(Types.scala:5697)
[error] dotty.tools.dotc.core.Types$TypeMap.mapOver(Types.scala:5779)
[error] dotty.tools.dotc.core.Types$TypeMap.mapOver(Types.scala:5852)
[error] dotty.tools.dotc.typer.ImplicitRunInfo$$anon$1.apply(Implicits.scala:800)
[error] dotty.tools.dotc.typer.ProtoTypes$ViewProto.map(ProtoTypes.scala:620)
[error] dotty.tools.dotc.typer.ProtoTypes$ViewProto.map(ProtoTypes.scala:620)
[error] dotty.tools.dotc.core.Types$TypeMap.mapOver(Types.scala:5803)
[error] dotty.tools.dotc.typer.ImplicitRunInfo$$anon$1.apply(Implicits.scala:800)
[error] dotty.tools.dotc.typer.ImplicitRunInfo.implicitScope(Implicits.scala:802)
[error] dotty.tools.dotc.typer.ImplicitRunInfo.implicitScope$(Implicits.scala:584)
[error] dotty.tools.dotc.Run.implicitScope(Run.scala:36)
[error] dotty.tools.dotc.typer.Implicits$ImplicitSearch.implicitScope(Implicits.scala:1576)
[error] dotty.tools.dotc.typer.Implicits$ImplicitSearch.searchImplicit(Implicits.scala:1539)
[error] dotty.tools.dotc.typer.Implicits$ImplicitSearch.searchImplicit(Implicits.scala:1548)
[error] dotty.tools.dotc.typer.Implicits$ImplicitSearch.bestImplicit(Implicits.scala:1573)
[error] dotty.tools.dotc.typer.Implicits.inferImplicit(Implicits.scala:1061)
[error] dotty.tools.dotc.typer.Implicits.inferImplicit$(Implicits.scala:819)
[error] dotty.tools.dotc.typer.Typer.inferImplicit(Typer.scala:117)
[error] dotty.tools.dotc.typer.Implicits.inferView(Implicits.scala:857)
[error] dotty.tools.dotc.typer.Implicits.inferView$(Implicits.scala:819)
[error] dotty.tools.dotc.typer.Typer.inferView(Typer.scala:117)
[error] dotty.tools.dotc.typer.Typer.adaptToSubType$1(Typer.scala:4168)
[error] dotty.tools.dotc.typer.Typer.adaptNoArgsOther$1(Typer.scala:3996)
[error] dotty.tools.dotc.typer.Typer.adaptNoArgs$1(Typer.scala:4071)
[error] dotty.tools.dotc.typer.Typer.adapt1(Typer.scala:4277)
[error] dotty.tools.dotc.typer.Typer.adapt(Typer.scala:3590)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.typedArg(ProtoTypes.scala:497)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:897)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:897)
[error] dotty.tools.dotc.typer.Applications$Application.$anonfun$5$$anonfun$1(Applications.scala:643)
[error] scala.collection.immutable.List.map(List.scala:246)
[error] dotty.tools.dotc.typer.Applications$Application.$anonfun$5(Applications.scala:643)
[error] dotty.tools.dotc.typer.Applications.harmonic(Applications.scala:2333)
[error] dotty.tools.dotc.typer.Applications.harmonic$(Applications.scala:352)
[error] dotty.tools.dotc.typer.Typer.harmonic(Typer.scala:117)
[error] dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:645)
[error] dotty.tools.dotc.typer.Applications$Application.init(Applications.scala:492)
[error] dotty.tools.dotc.typer.Applications$TypedApply.<init>(Applications.scala:779)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.<init>(Applications.scala:896)
[error] dotty.tools.dotc.typer.Applications.ApplyTo(Applications.scala:1126)
[error] dotty.tools.dotc.typer.Applications.ApplyTo$(Applications.scala:352)
[error] dotty.tools.dotc.typer.Typer.ApplyTo(Typer.scala:117)
[error] dotty.tools.dotc.typer.Applications.simpleApply$1(Applications.scala:969)
[error] dotty.tools.dotc.typer.Applications.realApply$1$$anonfun$2(Applications.scala:1052)
[error] dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:3327)
[error] dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:1063)
[error] dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:1101)
[error] dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:352)
[error] dotty.tools.dotc.typer.Typer.typedApply(Typer.scala:117)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3050)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3115)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.$anonfun$7(ProtoTypes.scala:495)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.cacheTypedArg(ProtoTypes.scala:418)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.typedArg(ProtoTypes.scala:496)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:897)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:897)
[error] dotty.tools.dotc.typer.Applications$Application.addTyped$1(Applications.scala:589)
[error] dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:653)
[error] dotty.tools.dotc.typer.Applications$Application.init(Applications.scala:492)
[error] dotty.tools.dotc.typer.Applications$TypedApply.<init>(Applications.scala:779)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.<init>(Applications.scala:896)
[error] dotty.tools.dotc.typer.Applications.ApplyTo(Applications.scala:1126)
[error] dotty.tools.dotc.typer.Applications.ApplyTo$(Applications.scala:352)
[error] dotty.tools.dotc.typer.Typer.ApplyTo(Typer.scala:117)
[error] dotty.tools.dotc.typer.Applications.simpleApply$1(Applications.scala:969)
[error] dotty.tools.dotc.typer.Applications.realApply$1$$anonfun$2(Applications.scala:1052)
[error] dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:3327)
[error] dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:1063)
[error] dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:1101)
[error] dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:352)
[error] dotty.tools.dotc.typer.Typer.typedApply(Typer.scala:117)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3050)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3115)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.$anonfun$7(ProtoTypes.scala:495)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.cacheTypedArg(ProtoTypes.scala:418)
[error] dotty.tools.dotc.typer.ProtoTypes$FunProto.typedArg(ProtoTypes.scala:496)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:897)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.typedArg(Applications.scala:897)
[error] dotty.tools.dotc.typer.Applications$Application.$anonfun$5$$anonfun$1(Applications.scala:643)
[error] scala.collection.immutable.List.map(List.scala:246)
[error] dotty.tools.dotc.typer.Applications$Application.$anonfun$5(Applications.scala:643)
[error] dotty.tools.dotc.typer.Applications.harmonic(Applications.scala:2333)
[error] dotty.tools.dotc.typer.Applications.harmonic$(Applications.scala:352)
[error] dotty.tools.dotc.typer.Typer.harmonic(Typer.scala:117)
[error] dotty.tools.dotc.typer.Applications$Application.matchArgs(Applications.scala:645)
[error] dotty.tools.dotc.typer.Applications$Application.init(Applications.scala:492)
[error] dotty.tools.dotc.typer.Applications$TypedApply.<init>(Applications.scala:779)
[error] dotty.tools.dotc.typer.Applications$ApplyToUntyped.<init>(Applications.scala:896)
[error] dotty.tools.dotc.typer.Applications.ApplyTo(Applications.scala:1126)
[error] dotty.tools.dotc.typer.Applications.ApplyTo$(Applications.scala:352)
[error] dotty.tools.dotc.typer.Typer.ApplyTo(Typer.scala:117)
[error] dotty.tools.dotc.typer.Applications.simpleApply$1(Applications.scala:969)
[error] dotty.tools.dotc.typer.Applications.realApply$1$$anonfun$2(Applications.scala:1052)
[error] dotty.tools.dotc.typer.Typer.tryEither(Typer.scala:3327)
[error] dotty.tools.dotc.typer.Applications.realApply$1(Applications.scala:1063)
[error] dotty.tools.dotc.typer.Applications.typedApply(Applications.scala:1101)
[error] dotty.tools.dotc.typer.Applications.typedApply$(Applications.scala:352)
[error] dotty.tools.dotc.typer.Typer.typedApply(Typer.scala:117)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3050)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3115)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3187)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3191)
[error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3303)
[error] dotty.tools.dotc.typer.Namer.typedAheadExpr$$anonfun$1(Namer.scala:1656)
[error] dotty.tools.dotc.typer.Namer.typedAhead(Namer.scala:1646)
[error] dotty.tools.dotc.typer.Namer.typedAheadExpr(Namer.scala:1656)
[error] dotty.tools.dotc.typer.Namer.typedAheadRhs$1$$anonfun$1(Namer.scala:1909)
[error] dotty.tools.dotc.inlines.PrepareInlineable$.dropInlineIfError(PrepareInlineable.scala:243)
[error] dotty.tools.dotc.typer.Namer.typedAheadRhs$1(Namer.scala:1909)
[error] dotty.tools.dotc.typer.Namer.rhsType$1(Namer.scala:1917)
[error] dotty.tools.dotc.typer.Namer.cookedRhsType$1(Namer.scala:1935)
[error] dotty.tools.dotc.typer.Namer.lhsType$1(Namer.scala:1936)
[error] dotty.tools.dotc.typer.Namer.inferredResultType(Namer.scala:1947)
[error] dotty.tools.dotc.typer.Namer.inferredType$1(Namer.scala:1694)
[error] dotty.tools.dotc.typer.Namer.valOrDefDefSig(Namer.scala:1701)
[error] dotty.tools.dotc.typer.Namer$Completer.typeSig(Namer.scala:787)
[error] dotty.tools.dotc.typer.Namer$Completer.completeInCreationContext(Namer.scala:934)
[error] dotty.tools.dotc.typer.Namer$Completer.complete(Namer.scala:814)
[error] dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:174)
[error] dotty.tools.dotc.core.Denotations$Denotation.completeInfo$1(Denotations.scala:187)
[error] dotty.tools.dotc.core.Denotations$Denotation.info(Denotations.scala:189)
[error] dotty.tools.dotc.core.SymDenotations$SymDenotation.ensureCompleted(SymDenotations.scala:393)
[error] dotty.tools.dotc.typer.Typer.retrieveSym(Typer.scala:2991)
[error] dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3016)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3114)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3187)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3191)
[error] dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3213)
[error] dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3259)
[error] dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:2669)
[error] dotty.tools.dotc.typer.Typer.typedTypeOrClassDef$1(Typer.scala:3038)
[error] dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3042)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3114)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3187)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3191)
[error] dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3213)
[error] dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3259)
[error] dotty.tools.dotc.typer.Typer.typedBlockStats(Typer.scala:1161)
[error] dotty.tools.dotc.typer.Typer.typedBlock(Typer.scala:1165)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3058)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3115)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3187)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3191)
[error] dotty.tools.dotc.typer.Typer.typedNew(Typer.scala:913)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3054)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3115)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3187)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3191)
[error] dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3240)
[error] dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3259)
[error] dotty.tools.dotc.typer.Typer.typedClassDef(Typer.scala:2669)
[error] dotty.tools.dotc.typer.Typer.typedTypeOrClassDef$1(Typer.scala:3038)
[error] dotty.tools.dotc.typer.Typer.typedNamed$1(Typer.scala:3042)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3114)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3187)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3191)
[error] dotty.tools.dotc.typer.Typer.traverse$1(Typer.scala:3213)
[error] dotty.tools.dotc.typer.Typer.typedStats(Typer.scala:3259)
[error] dotty.tools.dotc.typer.Typer.typedPackageDef(Typer.scala:2812)
[error] dotty.tools.dotc.typer.Typer.typedUnnamed$1(Typer.scala:3083)
[error] dotty.tools.dotc.typer.Typer.typedUnadapted(Typer.scala:3115)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3187)
[error] dotty.tools.dotc.typer.Typer.typed(Typer.scala:3191)
[error] dotty.tools.dotc.typer.Typer.typedExpr(Typer.scala:3303)
[error] dotty.tools.dotc.typer.TyperPhase.typeCheck$$anonfun$1(TyperPhase.scala:44)
[error] dotty.tools.dotc.typer.TyperPhase.typeCheck$$anonfun$adapted$1(TyperPhase.scala:50)
[error] scala.Function0.apply$mcV$sp(Function0.scala:42)
[error] dotty.tools.dotc.core.Phases$Phase.monitor(Phases.scala:440)
[error] dotty.tools.dotc.typer.TyperPhase.typeCheck(TyperPhase.scala:50)
[error] dotty.tools.dotc.typer.TyperPhase.runOn$$anonfun$3(TyperPhase.scala:84)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] scala.collection.immutable.List.foreach(List.scala:333)
[error] dotty.tools.dotc.typer.TyperPhase.runOn(TyperPhase.scala:84)
[error] dotty.tools.dotc.Run.runPhases$1$$anonfun$1(Run.scala:246)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
[error] dotty.tools.dotc.Run.runPhases$1(Run.scala:262)
[error] dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:270)
[error] dotty.tools.dotc.Run.compileUnits$$anonfun$adapted$1(Run.scala:279)
[error] dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:71)
[error] dotty.tools.dotc.Run.compileUnits(Run.scala:279)
[error] dotty.tools.dotc.Run.compileSources(Run.scala:194)
[error] dotty.tools.dotc.Run.compile(Run.scala:179)
[error] dotty.tools.dotc.Driver.doCompile(Driver.scala:37)
[error] dotty.tools.xsbt.CompilerBridgeDriver.run(CompilerBridgeDriver.java:136)
[error] dotty.tools.xsbt.CompilerBridge.run(CompilerBridge.java:22)
[error] sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:91)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$7(MixedAnalyzingCompiler.scala:194)
[error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:249)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:184)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4$adapted(MixedAnalyzingCompiler.scala:164)
[error] sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:239)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:164)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:212)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:534)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:534)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$3(Incremental.scala:178)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$3$adapted(Incremental.scala:176)
[error] sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:454)
[error] sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:117)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
[error] sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:265)
[error] sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:409)
[error] sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:496)
[error] sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:396)
[error] sbt.internal.inc.Incremental$.apply(Incremental.scala:170)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:534)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:488)
[error] sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:425)
[error] sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
[error] sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:2427)
[error] sbt.Defaults$.$anonfun$compileIncrementalTask$2(Defaults.scala:2377)
[error] sbt.internal.server.BspCompileTask$.$anonfun$compute$1(BspCompileTask.scala:41)
[error] sbt.internal.io.Retry$.apply(Retry.scala:47)
[error] sbt.internal.io.Retry$.apply(Retry.scala:29)
[error] sbt.internal.io.Retry$.apply(Retry.scala:24)
[error] sbt.internal.server.BspCompileTask$.compute(BspCompileTask.scala:41)
[error] sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:2375)
[error] scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
[error] sbt.std.Transform$$anon$4.work(Transform.scala:69)
[error] sbt.Execute.$anonfun$submit$2(Execute.scala:283)
[error] sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
[error] sbt.Execute.work(Execute.scala:292)
[error] sbt.Execute.$anonfun$submit$1(Execute.scala:283)
[error] sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[error] java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[error] java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[error] java.base/java.lang.Thread.run(Thread.java:833)
```
</details>

--- a/compiler/src/dotty/tools/dotc/core/TypeErrors.scala
+++ b/compiler/src/dotty/tools/dotc/core/TypeErrors.scala
@@ -73,6 +73,8 @@ class MissingType(val pre: Type, val name: Name)(using Context) extends TypeErro
       case _ if givenSelf.exists && givenSelf.member(name).exists =>
         i"""$name exists as a member of the self type $givenSelf of $cls
            |but it cannot be called on a receiver whose type does not extend $cls"""
+      case _ if pre.baseClasses.exists(_.findMember(name, pre, Private, EmptyFlags).exists) =>
+        i"$name is a private member in a base class"
       case _ =>
         missingClassFile
 

--- a/compiler/src/dotty/tools/dotc/reporting/messages.scala
+++ b/compiler/src/dotty/tools/dotc/reporting/messages.scala
@@ -3292,14 +3292,14 @@ object UnusedSymbol {
 
 class NonNamedArgumentInJavaAnnotation(using Context) extends SyntaxMsg(NonNamedArgumentInJavaAnnotationID):
 
-  override protected def msg(using Context): String = 
+  override protected def msg(using Context): String =
     "Named arguments are required for Java defined annotations"
     + Message.rewriteNotice("This", version = SourceVersion.`3.6-migration`)
 
-  override protected def explain(using Context): String = 
+  override protected def explain(using Context): String =
     i"""Starting from Scala 3.6.0, named arguments are required for Java defined annotations.
-        |Java defined annotations don't have an exact constructor representation 
-        |and we previously relied on the order of the fields to create one. 
+        |Java defined annotations don't have an exact constructor representation
+        |and we previously relied on the order of the fields to create one.
         |One possible issue with this representation is the reordering of the fields.
         |Lets take the following example:
         |

--- a/compiler/src/dotty/tools/dotc/typer/Implicits.scala
+++ b/compiler/src/dotty/tools/dotc/typer/Implicits.scala
@@ -821,6 +821,10 @@ trait ImplicitRunInfo:
           override def stopAt = StopAt.Static
           private val seen = util.HashSet[Type]()
 
+          override def derivedTypeBounds(tp: TypeBounds, lo: Type, hi: Type): Type =
+            if lo.exists && hi.exists then super.derivedTypeBounds(tp, lo, hi)
+            else NoType // Survive inaccessible types, for instance in i21543.scala.
+
           def applyToUnderlying(t: TypeProxy) =
             if seen.contains(t) then
               WildcardType

--- a/tests/neg/i20554-a.check
+++ b/tests/neg/i20554-a.check
@@ -7,8 +7,8 @@
   | Explanation (enabled by `-explain`)
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   | Starting from Scala 3.6.0, named arguments are required for Java defined annotations.
-  | Java defined annotations don't have an exact constructor representation 
-  | and we previously relied on the order of the fields to create one. 
+  | Java defined annotations don't have an exact constructor representation
+  | and we previously relied on the order of the fields to create one.
   | One possible issue with this representation is the reordering of the fields.
   | Lets take the following example:
   |
@@ -29,8 +29,8 @@
   | Explanation (enabled by `-explain`)
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   | Starting from Scala 3.6.0, named arguments are required for Java defined annotations.
-  | Java defined annotations don't have an exact constructor representation 
-  | and we previously relied on the order of the fields to create one. 
+  | Java defined annotations don't have an exact constructor representation
+  | and we previously relied on the order of the fields to create one.
   | One possible issue with this representation is the reordering of the fields.
   | Lets take the following example:
   |

--- a/tests/neg/i20554-b.check
+++ b/tests/neg/i20554-b.check
@@ -7,8 +7,8 @@
   | Explanation (enabled by `-explain`)
   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   | Starting from Scala 3.6.0, named arguments are required for Java defined annotations.
-  | Java defined annotations don't have an exact constructor representation 
-  | and we previously relied on the order of the fields to create one. 
+  | Java defined annotations don't have an exact constructor representation
+  | and we previously relied on the order of the fields to create one.
   | One possible issue with this representation is the reordering of the fields.
   | Lets take the following example:
   |

--- a/tests/neg/i21543.check
+++ b/tests/neg/i21543.check
@@ -1,0 +1,22 @@
+-- [E007] Type Mismatch Error: tests/neg/i21543.scala:10:15 ------------------------------------------------------------
+10 |      Cmd(List("1", "2")) // error // error
+   |               ^^^
+   |               Found:    ("1" : String)
+   |               Required: Event
+   |
+   |               Note that I could not resolve reference Event.
+   |               Event is a private member in a base class
+   |                                
+   |
+   | longer explanation available when compiling with `-explain`
+-- [E007] Type Mismatch Error: tests/neg/i21543.scala:10:20 ------------------------------------------------------------
+10 |      Cmd(List("1", "2")) // error // error
+   |                    ^^^
+   |                    Found:    ("2" : String)
+   |                    Required: Event
+   |
+   |                    Note that I could not resolve reference Event.
+   |                    Event is a private member in a base class
+   |                                     
+   |
+   | longer explanation available when compiling with `-explain`

--- a/tests/neg/i21543.scala
+++ b/tests/neg/i21543.scala
@@ -7,7 +7,7 @@ object CompilerCrash {
 
   new Scope {
     val commands = List(
-      Cmd(List("1", "2"))
+      Cmd(List("1", "2")) // error // error
     )
   }
 }

--- a/tests/neg/i21543.scala
+++ b/tests/neg/i21543.scala
@@ -1,0 +1,13 @@
+object CompilerCrash {
+  trait Scope {
+    private type Event = String
+
+    case class Cmd(events: List[Event])
+  }
+
+  new Scope {
+    val commands = List(
+      Cmd(List("1", "2"))
+    )
+  }
+}

